packages/date-picker/src/shared/testutils/getProviderPropsFromStoryContext/index.ts

```diff
@@ -3,36 +3,34 @@ import { StoryContext } from '@storybook/react';
 import { LeafyGreenProviderProps } from '@leafygreen-ui/leafygreen-provider';
 import { pickAndOmit } from '@leafygreen-ui/lib';
 
+import { DatePickerProps } from '../../../DatePicker/DatePicker.types';
 import {
   ContextPropKeys,
   contextPropNames,
   SharedDatePickerProviderProps,
 } from '../../context';
-import { BaseDatePickerProps } from '../../types';
 
 export interface ProviderPropsObject<T> {
   leafyGreenProviderProps: LeafyGreenProviderProps;
   datePickerProviderProps: SharedDatePickerProviderProps;
   storyProps: T;
 }
 
-export const getProviderPropsFromStoryContext = <P = BaseDatePickerProps>(
+export const getProviderPropsFromStoryContext = <P = DatePickerProps>(
   ctx: StoryContext<Partial<P & SharedDatePickerProviderProps>>,
-): ProviderPropsObject<Partial<Omit<P, ContextPropKeys>>> => {
+  // ): ProviderPropsObject<Partial<Omit<P, ContextPropKeys>>>
+): ProviderPropsObject<Partial<Omit<DatePickerProps, ContextPropKeys>>> => {
```

Comment:
Using `Partial<Omit<p, ContextPropKeys>>` shows an error on `storyProps` on L36 below, 
```
Type '{ [x: string]: any; [x: number]: any; }' is not assignable to type 'Partial<Omit<P, "aria-label".... 
```

If i change this to `Partial<Omit<DatePickerProps, ContextPropKeys>>` instead of `Partial<Omit<p, ContextPropKeys>>` the error goes away.

Reply:
If you pass in `ctx.args` (instead of just `ctx`) on L26, then `storyProps` should infer the correct type